### PR TITLE
Update deps (minor) on eslint-config-fluid in the client release group

### PR DESCRIPTION
**Pull request by jikim-msft**
### Description

This PR is one of the 4 PRs to update dependencies on `eslint-config-fluid` in the `client` Release group. After resetting the branch version to main's version, installed a new `flub` package and ran the following commands:
- `npm i -g @fluid-tools/build-cli@next`
- `flub bump deps @fluidframework/eslint-config-fluid -t latest -v -g client --no-commit`

Ran `npm run prettier:fix && npm run eslint:fix` inside the `package/dds/tree` directory to fix the prettier error.

### CODE REVIEW

File: api-report/experimental-tree.api.md [hunk 1]
```diff
@@ -652,7 +652,7 @@ export interface OrderedEditSet<TChange = unknown> {
     // (undocumented)
     tryGetEditAtIndex(index: number): Edit<TChange> | undefined;
     // (undocumented)
-    tryGetEditFromIdentifier(editId: EditId): Edit<TChange> | undefined;
+    tryGetEditFromId(editId: EditId): Edit<TChange> | undefined;
```
Comment (tylerbutler) [inline api-report/experimental-tree.api.md:655]: This change looks unrelated to the lint package upgrade.

**Review by tylerbutler — APPROVED** [review]
One minor unrelated change (I think), but other than that looks good! Feel free to merge once CI passes.

File: api-report/experimental-tree.api.md (hunk 1 above)
Comment (jikim-msft) [reply api-report/experimental-tree.api.md]: Just made the change! Thank you!
